stl/inc/chrono

```diff
@@ -2105,8 +2105,9 @@ namespace chrono {
 
     template <class _Ty>
     _NODISCARD const _Ty* _Locate_zone_impl(const vector<_Ty>& _Vec, string_view _Name) {
-        const auto _Result = _STD find_if(_Vec.begin(), _Vec.end(), [&](auto& _Tz) { return _Tz.name() == _Name; });
-        return _Result == _Vec.end() ? nullptr : &*_Result;
+        const auto _Result = _STD lower_bound(

```

Comment:
@cpplearner explained why this is possible in #5547, but (1) I think we should cite the Standardese here, and (2) the fact that ICU happens to be providing this sorting but we haven't commented or even checked that in the STL is scary. We should add test coverage.